### DISCUSSION TITLE
Accessing The Mesh Block IDs From Within A Kernel/Material

### DISCUSSION THREAD

**Discussion by pharshaad** [idaholab/moose#18155] [Q&A Modules: General] [answered]
Hi All,

I was wondering how I'd be able to access different mesh block IDs inside a custom kernel or a material. I have a microstructure mesh file (.msh) in which the grains are treated as subdomains/blocks. What I'd like to do is to able to get block IDs and apply the Euler angles to them. Any insight into this is greatly appreciated!

Thanks,
Ari

**Comment by GiudGiud** [comment, marked as answer]
Hi Leon

You can get the mesh block id from the element the kernel is currently working on with _current_elem->subdomain_id().
For the material it should be the same thing, looking briefly at the source code, though I have never used that. 

Best,
Guillaume

**Reply by pharshaad** [reply]
Thanks for the help, Guillaume. Much appreciated!
